### ISSUE TITLE
[B62/P01] Add Daum Search as primary grounded web provider

### ISSUE THREAD

**Issue by skerishKang** [skerishKang/ai-revenue-lab#1302]
## Owner direction

Padiem Chat is a public product, so hallucination control must not depend on a low-free-tier crawler/search provider. Prefer aggressive factual grounding while keeping search cost bounded.

Current candidate direction:

```text
PRIMARY_WEB_SEARCH = Daum Search API
GLOBAL/SECONDARY_FUTURE = Brave or equivalent
FIRECRAWL = optional complex-page extractor only, not default search
```

## Why Daum

Fresh Kakao Developers docs show:
- REST web search endpoint: `GET https://dapi.kakao.com/v2/search/web`
- results include `title`, `contents`, `url`, `datetime`
- Daum Search daily quota: 50,000 calls; web-document search daily quota: 30,000 calls
- quota increases are available through consultation/partnership

Kakao DevTalk staff have also previously stated that Daum Search API may be used within quota regardless of commercial/non-commercial purpose, and that storing/processing returned `title/contents/url` data is acceptable. No Daum Search documentation reviewed so far contains the explicit AI/LLM-input prohibition that currently makes Naver Search unsuitable for this grounding use case.

## Important legal/terms gate

Do **not** assume that "AI is the consumer" automatically grants reuse rights.

Before production activation, obtain/retain a written Kakao/DevTalk confirmation for this exact use:

> Padiem Chat sends Daum web-search result fields (`title`, `contents`, `url`, `datetime`) as grounded context to an LLM, which generates a user-facing answer with source attribution. Is this permitted for a public/commercial service within the provided quota?

Implementation may proceed with mocked/network-free contract tests before this production-activation confirmation.

## Required architecture

P01/Core remains web-provider authority. Do not implement a second search runtime in B62.

Add a Core-backed Daum provider contract with:
- provider id `daum`
- server-side REST API key only; never expose to browser
- bounded query/result sizes
- `accuracy` and `recency` sort support
- normalized evidence (`title`, `url`, `snippet/contents`, `retrieved_at/datetime`, provider=`daum`)
- URL safety policy preserved
- timeout/response-size bounds preserved
- no hidden fallback to model memory for MUST_SEARCH
- optional future provider routing without hard-coding B62 to one vendor

## Search policy

Shift quality objective from "minimize searches" to "minimize unsupported external factual claims".

- NO_SEARCH: translation, rewrite, source-bound summarization, creative work, arithmetic/logical transforms, stable concept explanations when freshness is irrelevant.
- SEARCH_DEFAULT: external factual questions involving named entities, precise dates/numbers/statistics, products/services, recommendations/comparisons, factual verification.
- MUST_SEARCH: current/latest/today/recent, news, prices/rates/markets, versions, policies/laws, office-holders, schedules/availability/outages, explicit web verification.
- MUST_SEARCH failure: fail truthfully; never answer from model memory as if verified.

## Cost controls

Design for public usage:
- normalized-query evidence cache
- TTL by freshness class
- deduplicate equivalent/current queries where safe
- one bounded search request per ordinary grounded turn by default
- top-k bounds
- per-user/global quotas
- provider circuit breaker / 429 handling
- no Firecrawl call unless a selected page truly requires complex extraction

## Acceptance before production

```text
DAUM_PROVIDER_CONTRACT = PASS
SECRET_BROWSER_EXPOSURE = 0
NETWORK_FREE_TESTS = PASS
MUST_SEARCH_GROUNDED = PASS
SEARCH_DEFAULT_GROUNDED = PASS
NO_SEARCH_UNCHANGED = PASS
SEARCH_FAILURE_NO_MEMORY_FALLBACK = PASS
CACHE_CONTRACT = PASS or separately gated before public scale
KAKAO_LLM_GROUNDING_TERMS_CONFIRMATION = REQUIRED_FOR_LIVE
FIRECRAWL_REQUIRED_FOR_DEFAULT_SEARCH = NO
```

Refs #1298 #913.

**Comment by skerishKang** [comment]
Implementation status update after PR #1303:

```text
PR1303 = SQUASH_MERGED
CURRENT_MAIN = 72debdd85cb48b8108888e52b5dcf057cf0331cf
DAUM_PROVIDER_CODE = MERGED
EXACT_HEAD_CI = 20/20 SUCCESS
KAKAO_REAL_API_CALLS = 0
PRODUCTION_WEB_PROVIDER_CHANGE = NO
PRODUCTION_MUTATION = 0
```

Merged contract:
- Core-owned `daum` provider using fixed `https://dapi.kakao.com/v2/search/web`.
- server-only `PADIEM_CHAT_DAUM_REST_API_KEY`.
- evidence normalization and unsafe URL dropping.
- `accuracy` / `recency` support.
- ordinary Daum search requires no Firecrawl.
- page fetch remains fail-closed unless optional Firecrawl extractor is separately configured.

Issue remains OPEN because live activation still requires written Kakao/DevTalk confirmation for LLM grounding use and then a separately gated #913 production binding/deploy step.

**Comment by skerishKang** [comment]
OWNER HOLD — Daum live activation deferred.

Current main already contains the Daum provider contract (`72debdd85cb48b8108888e52b5dcf057cf0331cf`), but the owner has explicitly deferred Kakao/Daum key issuance and production activation until later.

```text
LOCAL_EXECUTION = STOP
PRODUCTION_MUTATION = 0
DAUM_API_KEY_ISSUANCE = DEFERRED
CLOUDFLARE_SECRET_MUTATION = 0
PADIEM_CHAT_WEB_PROVIDER_CHANGE = NO
REAL_DAUM_CALLS = 0
```

Keep #1302 OPEN as the activation/terms/key follow-up. Do not request or print credentials until the owner resumes this gate.

**Comment by skerishKang** [comment]
Architecture refinement after live Daum result-quality inspection:

Daum provider remains a search acquisition adapter, but **raw Daum top-N must not be passed directly to Laguna**. The reusable trust/relevance layer belongs in Padiem AI Core (#1101), not B62.

Updated intended flow:

```text
B62 search decision
→ Core WebProvider (Daum)
→ Core source trust/relevance/ranking gate
→ normalized selected Evidence
→ Laguna grounded synthesis
→ B62 citation presentation
```

The Core gate should prefer authoritative/primary sources for current factual questions, demote low-authority/community/SEO sources when they are not task-appropriate, preserve community sources for sentiment/experience tasks, dedupe URLs, weight freshness, detect conflicts, and fail truthfully when adequate evidence is unavailable.

Daum API connectivity has been manually proven by the owner (HTTP 200 and documents returned), but production key installation remains deferred. Do not activate live Daum in Padiem Chat until the shared source-quality gate and existing Kakao terms gate are accepted.

Refs #1101 #1298.

**Comment by skerishKang** [comment]
Prerequisite update: shared Core Source Trust + Relevance Gate is now merged.

```text
PR = #1311 MERGED
CURRENT_MAIN = f9ff7f81602138daa674811b9650bb7ffc86cf97
DAUM_PROVIDER_CODE = READY
CORE_SOURCE_QUALITY_GATE = READY
DAUM_REAL_API_CALLS = 0
DAUM_KEY_INSTALL = NO
PRODUCTION_WEB_PROVIDER_CHANGE = NO
```

Daum raw provider order will not be passed directly to synthesis: Core now filters/ranks by query relevance, bounded source tier/authority hints, contextual community policy and canonical URL dedupe, and fails closed when no relevant evidence remains.

Live activation remains HOLD until the Daum REST key is installed server-side and the separate Kakao LLM-grounding terms/live QA gates are satisfied.

**Comment by skerishKang** [comment]
Kakao/Daum terms gate — fresh policy review (2026-09-01)

Conclusion: **do not activate Daum-backed LLM grounding in Production yet.** I found no current Kakao document that expressly prohibits transient LLM grounding, but I also found no current written authorization for the exact use: sending Daum web-search result fields to an LLM and publishing a synthesized answer to users.

Fresh official/current evidence:
- Daum Search docs describe the web-search API and quota, but do not state an AI/LLM-input prohibition or an express LLM-grounding permission: https://developers.kakao.com/docs/ko/daum-search/common and https://developers.kakao.com/docs/en/daum-search/dev-guide
- Current Kakao Developers Operating Policy (effective 2025-03-04) prohibits caching Kakao-received data except to improve the app user experience and requires cached data to be kept current (Art. 5(20)); it also broadly restricts using/providing information obtained through Kakao services without prior Kakao approval (Art. 5(30)): https://developers.kakao.com/terms/ko/site-policies-20250304
- The same policy permits Kakao to restrict/terminate service for Art. 5 violations (Art. 8).
- A Kakao DevTalk staff answer from 2021 says Daum Book Search API commercial use is not restricted within quota and that DB caching for service UX can be acceptable if kept current, while API-function reselling is not allowed: https://devtalk.kakao.com/t/api/116762
- A recent 2026 Kakao DevTalk answer for **Local API** says search-result storage/backfill is prohibited and results should be called live when provided to users. This is a different API family, so it is not treated as controlling Daum Search, but it shows that reuse/storage policy is service-specific and should not be inferred from older/general precedent: https://devtalk.kakao.com/t/api/150597

Operational interpretation for Padiem Chat:
```text
COMMERCIAL_SERVICE_USE = SUPPORTIVE_PRECEDENT_BUT_NOT_EXACT_LLM_CONFIRMATION
TRANSIENT_LLM_GROUNDING = NO_EXPLICIT_PROHIBITION_FOUND
PUBLIC_SYNTHESIZED_LLM_ANSWER = NO_EXPLICIT_PERMISSION_FOUND
PERSISTENT_SEARCH_CORPUS / TRAINING_DATASET = DO_NOT_DO
LONG_TERM_RAW_RESULT_STORAGE = DO_NOT_DO_WITHOUT_CLEAR_PERMISSION
SHORT_CACHE_FOR_USER_EXPERIENCE = POLICY_CONDITIONED; KEEP_CURRENT
API_RESELLING = NO
LIVE_ACTIVATION = HOLD
```

Required written Kakao/DevTalk confirmation before live activation:

> Padiem Chat is a public/commercial AI chat service. For each user question, our server calls Daum Web Search API (`/v2/search/web`) within quota, temporarily sends returned `title`, `contents`, `url`, and `datetime` fields to an LLM as grounding context, and returns a synthesized answer with source links/attribution. We do not train or fine-tune models on Daum results, do not resell the Search API, do not expose the REST API key, and do not build a permanent search-results corpus. We may use a short-lived cache only to improve user experience and keep it current. Is this usage permitted under the current Kakao Developers terms/policies for a public/commercial service?

Keep #1302 OPEN. No key/Cloudflare/provider mutation until an affirmative written response is retained.

**Comment by skerishKang** [comment]
OWNER HOLD — Kakao/Daum LLM-grounding terms confirmation remains an explicit prerequisite for live activation.

Current disposition:

```text
DAUM_PROVIDER_CODE = READY
CORE_SOURCE_QUALITY_GATE = READY
DAUM_API_CONNECTIVITY = MANUALLY_PROVEN
KAKAO_LLM_GROUNDING_TERMS_CONFIRMATION = PENDING
DAUM_KEY_INSTALL_TO_PRODUCTION = NO
PADIEM_CHAT_WEB_PROVIDER_CHANGE = NO
CLOUDFLARE_SECRET_MUTATION = 0
PRODUCTION_MUTATION = 0
```

Do not activate Daum Search in public Padiem Chat until written Kakao/DevTalk confirmation is retained for this exact use: Daum Search result fields (`title`, `contents`, `url`, `datetime`) are supplied transiently as LLM grounding context to generate a user-facing answer with source attribution, without model training/fine-tuning, API resale, or a permanent search-result corpus.

Keep #1302 OPEN as the authority issue for terms confirmation, server-side key installation, and later live activation/QA.

**Comment by skerishKang** [comment]
Implementation-quality update: #1321 / PR #1323 is merged on main `4e7a476588fb8444e898ab19567506416a0c5c60`.

Daum-backed automatic grounding will now pass provider results through the existing Core #1308 Source Trust + Relevance Gate before model context assembly. Irrelevant/off-topic results are filtered and all-rejected results fail `no_evidence` rather than falling back to model memory.

This does **not** change the live gate:
```text
KAKAO_LLM_GROUNDING_TERMS_CONFIRMATION = PENDING
DAUM_KEY_INSTALL_TO_PRODUCTION = NO
CLOUDFLARE_SECRET_MUTATION = 0
PRODUCTION_MUTATION = 0
```
#1302 remains OPEN/HOLD for Kakao written terms confirmation before Production activation.

**Comment by skerishKang** [comment]
## Follow-up split for tomorrow

Dedicated terms-confirmation child issue created so this gate is not buried in the broader Daum provider issue:

- #1324 `[B62/P01] Obtain Kakao written confirmation for Daum → LLM grounding`

Target follow-up: `2026-09-02`.

Until #1324 is resolved with an explicit official Kakao answer:

```text
KAKAO_LLM_GROUNDING_TERMS_CONFIRMATION = PENDING
DAUM_PRODUCTION_KEY_INSTALL = NO
CLOUDFLARE_SECRET_MUTATION = NO
PADIEM_CHAT_WEB_PROVIDER_DAUM_LIVE = NO
PRODUCTION_DEPLOY_FOR_DAUM = NO
```

#1302 remains the parent implementation/live-activation issue; #1324 owns the written-permission gate.

**Comment by skerishKang** [comment]
CENTRAL acceptance reconciliation after B62/Core maintenance — 2026-09-01

Fresh source main:

```text
CURRENT_MAIN = 6491d168e4cdc147f9e4a07ac72b794b36244aff
```

Current #1302 implementation acceptance:

```text
DAUM_PROVIDER_CONTRACT = PASS (merged #1303)
SECRET_BROWSER_EXPOSURE = 0
NETWORK_FREE_TESTS = PASS
MUST_SEARCH_GROUNDED = PASS by current automatic-grounding contract
SEARCH_DEFAULT_GROUNDED = PASS by current automatic-grounding contract
NO_SEARCH_UNCHANGED = PASS
SEARCH_FAILURE_NO_MEMORY_FALLBACK = PASS
SOURCE_TRUST_RELEVANCE_GATE = PASS (#1311 + #1323 lineage)
RAW_DAUM_TOP_N_DIRECT_TO_SYNTHESIS = NO
FIRECRAWL_REQUIRED_FOR_DEFAULT_SEARCH = NO
```

Fresh source inspection confirms `padiem_ai_core.web_runtime.DaumWebProvider` currently performs bounded provider lookup with no search-result/evidence cache. #1302 explicitly permits `CACHE_CONTRACT = PASS or separately gated before public scale`, so the scale cache requirement is now separately tracked as:

```text
CACHE_PUBLIC_SCALE_GATE = #1354 OPEN
INITIAL_BOUNDED_LIVE_QA = NOT_BLOCKED_BY_CACHE_GATE IF #1324 IS RESOLVED AND OWNER EXPLICITLY AUTHORIZES NO-CACHE BOUNDED QA
BROAD_PUBLIC_SCALE_DAUM = HOLD UNTIL #1354
```

#1354 deliberately does not invent a TTL before Kakao provides any cache/retention guidance. It requires bounded normalized Evidence only, no raw provider payload, no permanent corpus, no training data, no secrets/user identity, and freshness-specific behavior.

Remaining #1302 blockers are therefore external/operational, not missing provider implementation:

```text
KAKAO_LLM_GROUNDING_TERMS_CONFIRMATION = PENDING (#1324)
DAUM_REST_KEY_PRODUCTION_INSTALL = NO
CLOUDFLARE_SECRET_MUTATION = 0
BOUNDED_LIVE_QA = NOT_RUN
PRODUCTION_ACTIVATION = HOLD
```

No credential request, real Daum call, or Production mutation was performed in this reconciliation.

**Comment by skerishKang** [comment]
Implemented — capability verified in main. Triage verdict: IMPLEMENTED (evidence anchor in #2018 batch table; sample set CTO-verified: PR #1585 merged 2026-09-02, 7 module files present, 44 reference folders present). Closed per CTO execution of the #2018 triage.
